lib/geoengineer/resource.rb

```diff
@@ -155,19 +157,27 @@ def build_individual_remote_resource
   end
 
   def matched_remote_resource
-    self.class.fetch_remote_resources.select { |r| r._geo_id == _geo_id }
+    self.class.fetch_remote_resources(fetch_provider).select { |r| r._geo_id == _geo_id }
+  end
+
+  def fetch_provider
+    environment&.find_provider(provider)
   end
 
-  def self.fetch_remote_resources
-    return @_rr_cache if @_rr_cache
-    @_rr_cache = _fetch_remote_resources
-                 .reject { |resource| _ignore_remote_resource?(resource) }
-                 .map { |resource| GeoEngineer::Resource.build(resource) }
+  def self.fetch_remote_resources(provider)
+    # The cache key is the provider
+    # no provider no resource
+    provider_id = provider&.terraform_id || DEFAULT_PROVIDER
+    @_rr_cache ||= {}
+    return @_rr_cache[provider_id] if @_rr_cache[provider_id]
+    @_rr_cache[provider_id] = _fetch_remote_resources(provider)
+                              .reject { |resource| _ignore_remote_resource?(resource) }
+                              .map { |resource| GeoEngineer::Resource.build(resource) }
   end
 
   # This method must be implemented for each resource type
   # it must return a list of hashes with at least the key
-  def self._fetch_remote_resources
+  def self._fetch_remote_resources(provider)
```

Comment:
Is it possible to have the provider passed in at resource instantiation? It seems a bit weird to have to pass in the provider when a resource should have belong to an environment which should have a provider.